### DISCUSSION TITLE
Disable function-signature-help

### DISCUSSION THREAD

**Discussion by gchape** [clojure-emacs/cider#3718] [Q&A] [answered]
With the given setup, I disable eldoc for lsp-mode; but, when I jack-in (cider-mode), eldoc shows documentation _on hover_ and a function signature on _cursor at-point_. 
 
 I included the configuration(114-115) from Cider documentation, but it had _no effect._ 
 
 **Is it possible to disable eldoc-mode completely?** 

![emacs-cider1](https://github.com/clojure-emacs/cider/assets/118893979/395443e0-d54b-4e51-9c45-b0141741e4ed)
![emacs-cider2](https://github.com/clojure-emacs/cider/assets/118893979/e78dd643-2d9e-4e5d-9125-c143c359e18f)

**Comment by rrudakov** [comment]
Maybe you could also try `(setopt cider-use-tooltips nil)`.

**Reply by gchape** [reply]
> Maybe you could also try `(setopt cider-use-tooltips nil)`.

`(setq! cider-use-tooltips nil)` addressed the onHover documentation, but the cursor at-point still shows a function signature, which is bothersome when it's long.

**Reply by rrudakov** [reply]
> but the cursor at-point still shows a function signature

In this case you should probably disable `eldoc-mode` for Clojure.

**Reply by vemv** [reply]
I accomplish this with `(remove-hook 'eldoc-documentation-functions #'cider-eldoc)`

**Reply by gchape** [reply]
> I accomplish this with `(remove-hook 'eldoc-documentation-functions #'cider-eldoc)`

It did not work; I also tried `(add-hook! 'clojure-mode-hook (lambda() (eldoc-mode -1)))` and it still shows a function signature. It blows my mind.

**Reply by vemv** [reply]
Do you have clojure-lsp set up? It could be it

**Reply by gchape** [reply]
> Do you have clojure-lsp set up? It could be it

Yes, I have. What do you suggest? Also, the weird thing is that if I doom/reload while I am jacked in cider, it stops showing.

**Reply by vemv** [reply]
Maybe there's a lsp member in `eldoc-documentation-functions`?

Else I'd ask in Clojurians Slack

**Comment by gchape** [comment, marked as answer]
One can remove `eldoc-documentation-functions` from cider using following snippet:

```
(use-package! cider
  :after eglot
  :preface
  (defun disable-cider-eldoc ()
    (setq-local eldoc-documentation-functions
                (remove #'cider-eldoc eldoc-documentation-functions))
    )
  :hook
  (cider-mode . disable-cider-eldoc)
  )

```

**Reply by vemv** [reply]
Happy that you got it! Thanks for documenting a solution.

**Reply by gchape** [reply]
> Happy that you got it! Thanks for documenting a solution.

With pleasure! 🤝
